### DISCUSSION TITLE
A way to add button beside select to add a new value to the select

### DISCUSSION THREAD

**Discussion by soca100100** [filamentphp/filament#10449] [General]
Hi,
I have a resource that has a 'select' that is filled out through the relation from a table. I would like to have a button beside that select to give the ability to add a new record to the database's table and to be visible in the select:

`                          Select::make('bill_leading_id')
                                ->relationship('bill_leading', 'bill_leading_number')
                                ->required()
                                ->searchable()
                                ->label('Bill of leading number'),`
so the button should create a new record and refresh the content of that Select once it is saved.

**Comment by medfrk** [comment]
> Hi, I have a resource that has a 'select' that is filled out through the relation from a table. I would like to have a button beside that select to give the ability to add a new record to the database's table and to be visible in the select:
> 
> ` Select::make('bill_leading_id') ->relationship('bill_leading', 'bill_leading_number') ->required() ->searchable() ->label('Bill of leading number'),` so the button should create a new record and refresh the content of that Select once it is saved.

You may define a custom form that can be used to create a new record and attach it to the BelongsTo relationship:
https://filamentphp.com/docs/3.x/forms/fields/select#creating-a-new-option-in-a-modal

**Reply by soca100100** [reply]
Thanks @medfrk ! I appreciate your help. I will give it a try. Thanks again ;)

**Reply by medfrk** [reply]
a Code example:
` Select::make('supplier_id')
                                                    ->relationship('supplier', 'supplier_name')
                                                    ->searchable()
                                                    ->preload()
                                                    ->createOptionForm([
                                                        TextInput::make('supplier_name')
                                                            ->required()
                                                            ->maxLength(255),
                                                        TextInput::make('supplier_email')
                                                            ->label('Email address')
                                                            ->email()
                                                            ->required()
                                                            ->maxLength(255),
                                                        TextInput::make('supplier_phone')
                                                            ->label('Phone number')
                                                            ->tel()
                                                            ->required(),
                                                        TextInput::make('city')
                                                            ->label('City')
                                                            ->required(),
                                                        TextInput::make('country')
                                                            ->label('Country')
                                                            ->required(),
                                                        Textarea::make('address')
                                                            ->label('Adress')
                                                            ->required(),
                                                    ])
                                                    ->required(),`

**Reply by soca100100** [reply]
Thanks! It works. You made my day! 👍

**Reply by medfrk** [reply]
You're welcome! I'm really glad to hear that it worked for you. If you have any more questions or need further assistance, feel free to ask. Have a great day!
